### DISCUSSION TITLE
Custom events in Node? Event Target confusion

### DISCUSSION THREAD

**Discussion by o-t-w** [nodejs/help#4874] [Q&A] [answered]
I am a bit confused by the documentation for Event Target.

In browsers you can create custom events. Can you do that in Node? I couldn't find anything relevant from googling. 
The [docs use a "foo" event as an example](https://nodejs.org/api/events.html#events_event_listener
). Is this a custom event?

I tried reading through this blog post from [NearForm](https://www.nearform.com/blog/node-js-and-the-struggles-of-being-an-eventtarget/) but it uses a "click" event as an example, which obviously is not an event that would ever happen in Node.js so needs to be artificially dispatched, which just confuses me even more as to why you would ever use a fake click event in Node.

Also, because Event Target is very similar to Event Emitter but added later, I have the question of whether Event Emitter is still what I should use most of the time, or whether Event Target can replace it?

There aren't many resources/guides/blog posts in general about Event Target in Node and reading MDN is very browser-focused.

**Comment by jasnell** [comment, marked as answer]
Yes, you can create your own custom Event objects such as...

```js
class FooEvent extends Event {
  constructor() { super('foo'); }
}

const et = new EventTarget();
et.dispatchEvent(new FooEvent())
```

With regards to EventTarget and EventEmitter, they definitely are very difficult things. The decision of which one to use depends on a number of factors. If your application is already heavily using EventEmitter, then I would just stick with that. If you have the option of designing or controlling the entire API and wish to have an approach that is compatible with browsers, deno, workers or other environments, then EventTarget would be ideal.
